src/aosm/azext_aosm/build_processors/helm_chart_processor.py

```diff
@@ -242,9 +242,12 @@ def _generate_artifact_profile(self) -> AzureArcKubernetesArtifactProfile:
         """
         logger.debug("Generating artifact profile for Helm chart input.")
         image_pull_secrets_values_paths: Set[str] = set()
```

Comment:
This is what the function updates according to Jacob. So we cannot remove this code

Reply:
Thanks for checking Jordan. Can we make this clearer in the code? Perhaps either rename the `matches` argument `image_pull_secrets_values_paths`, or doc it much more clearly?

Reply:
I had a thought on how to refactor this so we have a clear return value rather than the hard to read modification of a passed by reference variable.

Wrap the current recursive method in an outer method that sets up the initial variable, and returns a clear final value for `image_pull_secrets_values_paths`.

Expect it can wait till I'm back.